From 4bd4790382696a1b744a3607bb91d1522fe5e57c Mon Sep 17 00:00:00 2001
From: Sam Day <me@samcday.com>
Date: Sun, 4 Oct 2020 21:52:03 +0200
Subject: [PATCH] [daemon] simplify systemd service management

Unify the service for main dom0/domU daemons into one systemd unit.
---
 daemon/db-daemon.c           | 17 ++++++++++-------
 daemon/qubes-db-dom0.service | 12 ------------
 daemon/qubes-db-dom0.socket  |  6 ------
 daemon/qubes-db.service      |  3 +--
 4 files changed, 11 insertions(+), 27 deletions(-)
 delete mode 100644 daemon/qubes-db-dom0.service
 delete mode 100644 daemon/qubes-db-dom0.socket

diff --git a/daemon/db-daemon.c b/daemon/db-daemon.c
index 40f3093..b2cacaf 100644
--- a/daemon/db-daemon.c
+++ b/daemon/db-daemon.c
@@ -649,8 +649,8 @@ void close_server_socket(struct db_daemon_data *d) {
 #endif // !WIN32
 
 void usage(char *argv0) {
-    fprintf(stderr, "Usage: %s <remote-domid> [<remote-name>]\n", argv0);
-    fprintf(stderr, "       Give <remote-name> only in dom0\n");
+    fprintf(stderr, "Usage: %s [<remote-domid>] [<remote-name>]\n", argv0);
+    fprintf(stderr, "       Give <remote-domid> + <remote-name> only in dom0\n");
 }
 
 #ifdef WIN32
@@ -690,18 +690,21 @@ int fuzz_main(int argc, char **argv) {
     under_systemd = getenv("NOTIFY_SOCKET") != NULL;
 #endif
 
-    if (argc != 2 && argc != 3 && argc != 4) {
+    if (argc != 1 && argc != 3) {
         usage(argv[0]);
         exit(1);
     }
 
     memset(&d, 0, sizeof(d));
 
-    d.remote_domid = atoi(argv[1]);
-    if (argc >= 3 && strlen(argv[2]) > 0)
+    // Unless remote id + name is given, we assume we're connecting to dom0.
+    d.remote_domid = 0;
+    d.remote_name = "dom0";
+
+    if (argc == 3 && strlen(argv[2]) > 0) {
+        d.remote_domid = atoi(argv[1]);
         d.remote_name = argv[2];
-    else
-        d.remote_name = NULL;
+    }
 
     /* if not running under SystemD, fork and use pipe() to notify parent about
      * sucessful start */
diff --git a/daemon/qubes-db-dom0.service b/daemon/qubes-db-dom0.service
deleted file mode 100644
index edbaf5b..0000000
--- a/daemon/qubes-db-dom0.service
+++ /dev/null
@@ -1,12 +0,0 @@
-[Unit]
-Description=Qubes DB agent
-After=xenstored.service
-Requires=xenstored.service
-
-[Service]
-# Both agent and daemon for dom0
-ExecStart=/usr/sbin/qubesdb-daemon 0 dom0
-Type=notify
-
-[Install]
-WantedBy=multi-user.target
diff --git a/daemon/qubes-db-dom0.socket b/daemon/qubes-db-dom0.socket
deleted file mode 100644
index 699349c..0000000
--- a/daemon/qubes-db-dom0.socket
+++ /dev/null
@@ -1,6 +0,0 @@
-[Socket]
-ListenStream=/run/qubes/qubesdb.sock
-Symlinks=/run/qubes/qubesdb.dom0.sock
-
-[Install]
-WantedBy=sockets.target
diff --git a/daemon/qubes-db.service b/daemon/qubes-db.service
index 1862a6e..2d00936 100644
--- a/daemon/qubes-db.service
+++ b/daemon/qubes-db.service
@@ -4,8 +4,7 @@ After=xenstored.service
 Requires=xenstored.service
 
 [Service]
-# Remote Dom ID as an argument
-ExecStart=/usr/sbin/qubesdb-daemon 0
+ExecStart=/usr/sbin/qubesdb-daemon
 Type=notify
 
 [Install]
-- 
2.28.0

